app/src/main/java/org/dhis2/data/forms/dataentry/fields/section/SectionHolder.kt

```diff
@@ -1,140 +1,24 @@
-                val isEllipsized = it.getEllipsisCount(0) > 0
-                if (viewModel.hasToShowDescriptionIcon(isEllipsized)) {
-                    formBinding.descriptionIcon.visibility = VISIBLE
-                }
-            }
-        }
-
-        setShadows()
-    }
-
-    override fun onClick(v: View) {
-        if (viewModel.uid() != SectionViewModel.CLOSING_SECTION_UID) {
-            sectionProcessor.onNext(viewModel.uid())
-        }
-    }
-
-    private fun setShadows() {
-        val isSelected = viewModel.isOpen
-        if (isSelected) {
-            formBinding.shadowTop.visibility = VISIBLE
-        } else {
-            formBinding.shadowTop.visibility = GONE
-        }
-    }
-
-    private fun animateArrow() {
-        val isSelected = selectedSection.get() == viewModel.uid()
-        if (isSelected) {
-            formBinding.openIndicator.scaleY = 1f
-        }
-        formBinding.openIndicator.animate()
-            .scaleY(if (isSelected) 1f else -1f)
-            .setDuration(200)
-            .start()
-    }
-
-    fun setBottomShadow(showShadow: Boolean) {
-        formBinding.shadowBottom.visibility = if (showShadow) VISIBLE else GONE
-    }
-
-    fun setLastSectionHeight(previousSectionIsOpened: Boolean) {
-        val params = formBinding.lastSectionDetails.layoutParams
-        val finalHeight = if (previousSectionIsOpened) {
-            48.dp
-        } else {
-            1.dp
-        }
-        ValueAnimator.ofInt(params.height, finalHeight).apply {
-            duration = 120
-            addUpdateListener {
-                params.height = it.animatedValue as Int
-                formBinding.lastSectionDetails.layoutParams = params
-            }
-            start()
+        (viewModel as SectionViewModel).apply {
+            formBinding.sectionView.setViewModel(this)
         }
     }
 
-    private fun showDescription() {
-        CustomDialog(
-            itemView.context,
-            viewModel.label(),
-            viewModel.description() ?: "",
-            itemView.context.getString(R.string.action_close),
-            null,
-            201,
-            null
-        ).show()
-    }
-
-    fun handleHeaderClick(x: Float) {
-        val hasDescription = formBinding.descriptionIcon.visibility == VISIBLE
-        val descriptionClicked =
-            formBinding.descriptionIcon.x <= x &&
-                formBinding.descriptionIcon.x + formBinding.descriptionIcon.width >= x
-        if (hasDescription && descriptionClicked) {
-            showDescription()
-        } else {
-            onClick(itemView)
-        }
-    }
-
-    fun setSectionNumber(sectionNumber: Int) {
-        formBinding.sectionNumber.apply {
-            text = sectionNumber.toString()
-            background =
-                ContextCompat.getDrawable(itemView.context, R.drawable.ic_circle)
-        }
+    fun getSectionView(): SectionView {
+        return formBinding.sectionView
```

Comment:
in order to remove all the holders, could we think to remove this from here. Let's check it when everything is merged. Good job with this class :clap: